apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs

```diff
@@ -30,23 +29,30 @@ namespace Google.Cloud.Storage.V1.Tests;
 
     public static TheoryData<TestCase> TestCases { get; } = new TheoryData<TestCase>
     {
-        new TestCase("SimpleSuccess", RetryOptions.Never, new[] { OK }, new[] { 0 }, true),
-        new TestCase("ImmediateFailure", RetryOptions.IdempotentRetryOptions, new[] { BadRequest }, new[] { 0 }, false),
-        new TestCase("SingleRetry", RetryOptions.IdempotentRetryOptions, new[] { InternalServerError, OK }, new[] { 0, 1 }, true),
-        new TestCase("DoubleRetry", RetryOptions.IdempotentRetryOptions, new[] { InternalServerError, BadGateway, OK }, new[] { 0, 1, 3 }, true),
-        // We default to a maximum of 3 requests
+        new TestCase("SimpleSuccess_NoRetry", RetryOptions.IdempotentRetryOptions, new[] { OK }, new[] { 0 }, true),
+        new TestCase("SuccessAfterMaxRetry", RetryOptions.IdempotentRetryOptions, new[] { BadGateway, BadGateway, OK }, new[] { 0, 1, 3 }, true),
+        new TestCase("FailureAfterMaxRetry", RetryOptions.IdempotentRetryOptions, new[] { BadGateway, BadGateway, BadGateway }, new[] { 0, 1, 3 }, false),
+
+        new TestCase("CustomTiming_SuccessAfterRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, OK }, new[] { 0, 1, 3 }, true),
+        new TestCase("CustomTiming_FailureAfterRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, BadRequest }, new[] { 0, 1, 3}, false),
+
+        new TestCase("CustomPredicate_Success", s_customRetryOptions, new[] { OK }, new[] { 0 }, true),
+        new TestCase("CustomPredicate_ImmediateFailure", s_customRetryOptions, new[] { InternalServerError }, new[] { 0 }, false),
+        new TestCase("CustomPredicate_SingleRetry", s_customRetryOptions, new[] { NotFound, OK }, new[] { 0, 1 }, true),
+        new TestCase("CustomPredicate_DoubleRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, OK }, new[] { 0, 1, 3 }, true),
+        new TestCase("CustomPredicate_FailureAfterRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, BadRequest }, new[] { 0, 1, 3 }, false),
+        new TestCase("CustomPredicate_SuccessAfterRetry",  s_customRetryOptions, new[] { NotFound, OK }, new[] { 0, 1 }, true),
+        new TestCase("CustomPredicate_Never", RetryOptions.Never, new[] { BadGateway }, new[] { 0 }, false),
+
         new TestCase("RetryUntilMaxRequests", RetryOptions.IdempotentRetryOptions, new[] { BadGateway, BadGateway, BadGateway }, new[] { 0, 1, 3 }, false),
-        new TestCase("CustomPredicate_Success", s_customPredicateOptions, new[] { OK }, new[] { 0 }, true),
-        // We'd normally retry on this, but the custom predicate doesn't match.
-        new TestCase("CustomPredicate_ImmediateFailure", s_customPredicateOptions, new[] { InternalServerError }, new[] { 0 }, false),
-        new TestCase("CustomPredicate_SingleRetry", s_customPredicateOptions, new[] { NotFound, OK }, new[] { 0, 1 }, true)
+        new TestCase("RetryableErrors", s_customRetryOptions, new[] { BadRequest, BadRequest, OK }, new[] { 0, 1, 3 }, true),
```

Comment:
It's not clear to me why the RetryUntilMaxRequests is in amongst the custom options tests, rather than with all the other ones using RetryOptions.IdempotentRetryOptions at the top.